### PR TITLE
infra: notify merged PRs on close

### PR DESCRIPTION
**Pull request by ST-DDT**
This PR adds a notification to merged PRs (from the changelog) when a release with that change has been published.

This hasn't been tested in GH CI yet, but should be ready to review.

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -64,3 +64,16 @@ jobs:
       - name: Push to Release Branch
         run: |
           git push origin $GITHUB_REF_NAME:v$RELEASE_MAJOR
+
+      - name: Notify closed PRs
```
Comment (ST-DDT) [inline .github/workflows/publish-release.yml]:
```suggestion
      - name: Notify merged PRs
```

File: .github/workflows/publish-release.yml [hunk 2]
```diff
@@ -64,3 +64,16 @@ jobs:
...
+        run: |
+          VERSION_LINES=$(grep -n '^## ' CHANGELOG.md | head -n 2 | cut -d: -f1)
+          LATEST_VERSION_LINE=$(echo "$VERSION_LINES" | head -n 1)
+          PREVIOUS_VERSION_LINE=$(echo "$VERSION_LINES" | tail -n 1)
+          CHANGELOG_SECTION=$(sed -n "${LATEST_VERSION_LINE},$(($PREVIOUS_VERSION_LINE - 1))p" CHANGELOG.md)
```
Comment (ST-DDT) [inline .github/workflows/publish-release.yml]:
```suggestion
          CHANGELOG_SECTION=$(sed -n "$((LATEST_VERSION_LINE + 1)),$(($PREVIOUS_VERSION_LINE - 1))p" CHANGELOG.md)
```

**Comment by Shinigami92** [comment]
I think this is not necessary as GitHub already provides this out of the box

image
`example: https://github.com/faker-js/faker/pull/1855`

**Comment by ST-DDT** [comment]
The idea behind this is not showing everybody whether a feature is included or not, but telling the author of the PR (and anyone subscribed) that the feature is **now** available as a release.
Otherwise the user may not notice that there is a new release with their fix/feature unless the subscribed to faker-js releases or use renovate to update their dependencies.

> image

Tells me **in which version** something is available if you look for it. (Focus on version)

The comment tells me **when** something is available as soon as it is available. (Focus on timing)
Theoretically, you could omit the version number and just refer to the latest release, the version number is just for convenience.

This PR is based on:

- https://github.com/renovatebot/renovate/pull/30095#issuecomment-2257613647

> 🎉 This issue has been resolved in version 38.11.0 🎉

I hope this explains my intentions better.

@Shinigami92 @import-brain Do you still think this is redundant? If yes, then we can close this.

**Comment by Shinigami92** [comment]
You can watch/subscribe to issues/PRs
You can configure Renovate
You can read changelogs 

I think if you are really interested in a feature/bug being released, you are already affected and observe the sources
At least this is what I do

So IMO this is still just another push notifications that notifies me beside Renovate or VSCode version lens

All others are already potentially using not up-to-date code and life in peace 😅

**Comment by xDivisionByZerox** [comment]
> [...] I hope this explains my intentions better.

@ST-DDT I understand your reasoning and recognize the potential for positive feedback among contributors. However, I currently don't see a pressing need for this feature.

 > [...] At least this is what I do

@Shinigami92 please consider that your experience may not represent the majority of professionals in this field. But, I would argue that this actually supports your argument, as it implies that most developers may not stay as up-to-date as we are required to. I think you make a very good point that adding additional notifications could lead to notification fatigue, where users might start ignoring important updates due to the growing volume of notifications.

**Pull request closed without merging**